helm-chart/renku-core/templates/deployment.yaml

```diff
@@ -1,48 +1,53 @@
+{{- range $version := .Values.versions }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "renku-core.fullname" . }}
+  name: {{ include "renku-core.fullname" $ }}-{{ $version.name }}
   labels:
-{{ include "renku-core.labels" . | indent 4 }}
+    app.kubernetes.io/deploymentVersion: {{ $version.name }}
+{{ include "renku-core.labels" $ | indent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ $.Values.replicaCount }}
   strategy:
     type: Recreate
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "renku-core.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/name: {{ include "renku-core.name" $ }}
+      app.kubernetes.io/instance: {{ $.Release.Name }}
+      app.kubernetes.io/deploymentVersion: {{ $version.name }}
   template:
     metadata:
       annotations:
         prometheus.io/scrape: 'true'
         prometheus.io/path: '/metrics'
         prometheus.io/port: '8765'
       labels:
-        app.kubernetes.io/name: {{ include "renku-core.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
-        {{ include "call-nested" (list . "redis" "redis.fullname") }}-client: "true"
+        app.kubernetes.io/name: {{ include "renku-core.name" $ }}
+        app.kubernetes.io/instance: {{ $.Release.Name }}
+        app.kubernetes.io/deploymentVersion: {{ $version.name }}
+        {{ include "call-nested" (list $ "redis" "redis.fullname") }}-client: "true"
     spec:
-    {{- with .Values.imagePullSecrets }}
+    {{- with $.Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
     {{- end }}
       volumes:
         - name: shared-volume
           emptyDir: {}
       containers:
-      {{ if .Values.metrics.enabled }}
-        - name: {{ .Chart.Name}}-rqmetrics
-          image: "{{ .Values.metrics.image.repository }}:{{ .Values.metrics.image.tag }}"
-          imagePullPolicy: {{ .Values.metrics.image.pullPolicy }}
+      {{ if $.Values.metrics.enabled }}
```

Comment:
It would be nice to also apply a more restrictive security context here (same as for the new nginx deployment.yaml). This should be straight forward as the core service is already listening on port 8080.

Reply:
Setting DROP ALL causes some issues here and the service won't work as expected. I couldn't find what are the exact required capabilities since there is no logs/error messages. I removed DROP ALL from the core service.

Reply:
That's ok, for the moment making sure that containers run as non-root is the main concern that I have.